osquery/tables/system/windows/certificates.cpp

```diff
@@ -307,6 +311,230 @@ void parseSystemStoreString(LPCWSTR sysStoreW,
+      return Status::failure("Malformed certificate blob");
+    }
+
+    if (hdr.propid != CERT_CERT_PROP_ID) {
+      blob.ignore(hdr.size);
+      if (!blob.good()) {
+        return Status::failure("Malformed certificate blob");
+      }
+      continue;
+    }
+
+    encodedCert.resize(hdr.size);
+    blob.read(encodedCert.data(), hdr.size);
+    if (!blob.good()) {
+      return Status::failure("EOF in certificate blob when reading data");
+    }
+    break;
+  }
+
+  return Status::success();
+}
+
+void addCertRow(PCCERT_CONTEXT certContext,
+                QueryData& results,
+                std::string storeId,
+                std::string sid,
+                std::string storeName,
+                std::string username,
+                std::string storeLocation) {
+  std::vector<char> certBuff;
+  getCertCtxProp(certContext, CERT_HASH_PROP_ID, certBuff);
+  std::string fingerprint;
+  boost::algorithm::hex(std::string(certBuff.begin(), certBuff.end()),
+                        back_inserter(fingerprint));
+
+  Row r;
+  r["sid"] = sid;
+  r["username"] = username;
+  r["store_id"] = storeId;
+  r["sha1"] = fingerprint;
+  certBuff.resize(256, 0);
+  std::fill(certBuff.begin(), certBuff.end(), 0);
+  CertGetNameString(certContext,
+                    CERT_NAME_SIMPLE_DISPLAY_TYPE,
+                    0,
+                    nullptr,
+                    certBuff.data(),
+                    static_cast<unsigned long>(certBuff.size()));
+  r["common_name"] = certBuff.data();
+  TLOG << "    cert name: " << certBuff.data();
+
+  auto subjSize = CertNameToStr(certContext->dwCertEncodingType,
+                                &(certContext->pCertInfo->Subject),
+                                CERT_SIMPLE_NAME_STR,
+                                nullptr,
+                                0);
+  certBuff.resize(subjSize, 0);
+  std::fill(certBuff.begin(), certBuff.end(), 0);
+  subjSize = CertNameToStr(certContext->dwCertEncodingType,
+                           &(certContext->pCertInfo->Subject),
+                           CERT_SIMPLE_NAME_STR,
+                           certBuff.data(),
+                           subjSize);
+  r["subject"] = subjSize == 0 ? "" : certBuff.data();
+
+  auto issuerSize = CertNameToStr(certContext->dwCertEncodingType,
+                                  &(certContext->pCertInfo->Issuer),
+                                  CERT_SIMPLE_NAME_STR,
+                                  nullptr,
+                                  0);
+  certBuff.resize(issuerSize, 0);
+  std::fill(certBuff.begin(), certBuff.end(), 0);
+  issuerSize = CertNameToStr(certContext->dwCertEncodingType,
+                             &(certContext->pCertInfo->Issuer),
+                             CERT_SIMPLE_NAME_STR,
+                             certBuff.data(),
+                             issuerSize);
+  r["issuer"] = issuerSize == 0 ? "" : certBuff.data();
+
+  // TODO: Find the right API calls to get whether a cert is for a CA
```

Comment:
+1 on what @theopolis mentioned for `TODO`s, do you mind cutting a follow up issue for them and calling it out in the comment? You can maybe just create one "catch all" issue for small updates to this table and reference it here? Just want to make sure that's not left out.